frontend/src/components/Sidepanel/Panels/TestingLab/TestingLab.tsx

```diff
@@ -180,10 +180,16 @@ const TestingLab = () => {
 
   // Update disconnected warning
   const pathToFinal = useMemo(() => {
-    const closure = closureWithPredicate(graph, graph.initialState, () => true)
-    return Array.from(closure).some(({ state }) => graph.states.find(s => s.id === state)?.isFinal)
```

Comment:
```ts
    // Since closure doesn't add initial
    return Array.from(closure).concat(graph.initalState).some(({ state }) => graph.states.find(s => s.id === state)?.isFinal)
```
Wouldn't this be enough?

Reply:
Can then move the checking of final state existance down from 179 into something like this
```ts
if (noFinalState) { warnings.push('There are no final states') }
else if (!pathToFinal) { warnings.push('There is no path to final state') }
```
So that only one error is shown

Reply:
![image](https://github.com/automatarium/automatarium/assets/103556103/ceb6fa79-6a62-47d5-bb34-3f8ee9b7334c)
VSC is apparently not particularly happy with using _concat_ that way, but the idea is otherwise more sound than mine - I will look to restructure for this shortly.
You are right about the warnings though, that is a better way to do things (have fixed on my end, will look to push soon)

Reply:
Neat, that works. Thanks for that.

I'd also like to take a moment to appreciate the linter deciding to append it all to one line, from
`if (noFinalState) { warnings.push('There are no final states') }
else if (!pathToFinal) { warnings.push('There is no path to final state') }`
to
`if (noFinalState) { warnings.push('There are no final states') } else if (!pathToFinal) { warnings.push('There is no path to final state') }`

Reply:
Linter can be a lil silly at times